examples/notificationReload.c

```diff
@@ -0,0 +1,187 @@
+/**
+ * @file
+ *
+ * @brief Example for notification library which reloads KDB when Elektra's
+ * configuration (e.g. mount points or global plugins) has changed.
+ *
+ * Requires:
+ *   - io_glib binding
+ *   - Transport plugins (e.g. kdb global-mount zeromqsend zeromqrecv && kdb run-hub-zeromq)
+ *
+ * Ideas for this example:
+ *   - /sw/tests/example_notification/#0/current/value: Set to any integer value
+ *   - Try to add additional transport plugins and remove the original pair afterwards
+ *   - Mount a file which sets the key above to a different value and unmount it
+ *
+ * @copyright BSD License (see LICENSE.md or https://www.libelektra.org)
+ *
+ */
+
+#include <kdb.h>
+#include <kdbhelper.h>       // elektraFree
+#include <kdbio.h>	   // I/O binding functions (elektraIo*)
+#include <kdbio/glib.h>      // I/O binding constructor for glib (elektraIoGlibNew)
+#include <kdbnotification.h> // notification functions
+
+#include <glib-unix.h> // g_unix_signal_add()
+#include <glib.h>      // glib functions
+
+#include <signal.h> // signal()
+#include <stdio.h>  // printf() & co
+#include <stdlib.h> // exit()
+
+GMainLoop * loop;
+
+KDB * kdb;
+Key * parentKey;
+KeySet * config;
+ElektraIoInterface * binding;
+Key * intKeyToWatch;
+int valueToPrint = 0;
+
+ElektraIoTimerOperation * timer;
+ElektraIoTimerOperation * reload;
+
+static void elektraChangedCallback (Key * changedKey ELEKTRA_UNUSED, void * context ELEKTRA_UNUSED);
+
+/**
+ * Initializes KDB on first call and performs cleanup before initialization on
+ * subsequent calls.
+ *
+ * @param timerOp unused
+ */
+static void initKdb (ElektraIoTimerOperation * timerOp ELEKTRA_UNUSED)
+{
+	int didReload = 0;
+
+	// Stop reload task
+	elektraIoTimerSetEnabled (reload, 0);
+	elektraIoBindingUpdateTimer (reload);
+
+	if (kdb != NULL)
+	{
+		// Cleanup notifications and close KDB
+		elektraNotificationClose (kdb);
+		kdbClose (kdb, parentKey);
+		didReload = 1;
+	}
+
+	kdb = kdbOpen (parentKey);
+	if (kdb == NULL)
+	{
+		printf ("could not open KDB. aborting\n");
+		exit (-1);
```

Comment:
This would return "status & 0377" to the parent. Why not using exit (1)?

Reply:
No reason really, except the other examples used "return -1"; I'll use `exit (1)`